### ISSUE TITLE
SIM117 false negative for nested async context managers beneath a sync with parent

### ISSUE THREAD

**Issue by RomanValov** [astral-sh/ruff#27800]
### Summary

Please check the snippet below. SIM117 is supposed to be triggered in both cases of consecutive `async with` usage. However it is not triggered if consecutive `async with`s are wrapped by sync `with`:

```python
import asyncio
import contextlib


@contextlib.asynccontextmanager
async def mycontext():
    yield None


async def main():
    with contextlib.nullcontext() as ctx:
        # does not trigger SIM117
        async with mycontext() as actx1:
            async with mycontext() as actx2:
                print(ctx, actx1, actx2)

    # triggers SIM117
    async with mycontext() as actx1:
        async with mycontext() as actx2:
            print(ctx, actx1, actx2)


if __name__ == '__main__':
    asyncio.run(main())
```

Observed behavior:

```
$ ruff check --output-format concise ./context.py
context.py:17:5: SIM117 [*] Use a single `with` statement with multiple contexts instead of nested `with` statements
Found 1 error.
[*] 1 fixable with the `--fix` option.
```

Expected behavior:

```
Found 2 errors.
```

### Version

ruff 0.16.2

**Comment by ntBre** [comment]
Thanks for the report, this does seem like a bug. We should be able to flag the inner async context managers even if they can't be combined with the outer sync context manager.

We need to perform the same checks on the `with_parent` as on the child, otherwise we return too aggressively here:

https://github.com/astral-sh/ruff/blob/e986989cea9ca740d7b7ca2e298533ee64dafa5f/crates/ruff_linter/src/rules/flake8_simplify/rules/ast_with.rs#L151-L166
